### PR TITLE
Join room component

### PR DESCRIPTION
**Pull request by KmBarnett**
# Description :: User Story

## Type of change

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Breaking Change

## Notes
   This PR adds the functionality to join and create rooms
## Test results
 Passing

### CODE REVIEW

File: src/Components/RoomForm/RoomForm.js [hunk 1]
```diff
@@ -81,14 +81,14 @@ function RoomForm() {
...
-          </button>
+            <button className='cancel' type='button'>
+            Cancel
+            </button>
           </Link>
           <Link to='/'>
-          <button disabled={ !roomname } type='button'>
-          Create Group
-          </button>
+            <button disabled={ !groupName } type='button'>
```
Comment (Yetidancer) [inline src/Components/RoomForm/RoomForm.js]: naming change

File: src/Components/RoomForm/RoomForm.js [hunk 2]
```diff
@@ -42,43 +49,49 @@ function RoomForm() {
...
+          {createRules()}
+          <div className='empty-space' ref={messagesEndRef} />
+        </section>
+        <section className='rules-input'>
+          <input id='rule-input'
+          type='text'
+          placeholder='Rule'
+          value={ potentialRule }
+          onChange={ (e) => { setRule(e.target.value) } }
+          onKeyUp={ (e) => { handleEnter(e) } }
```
Comment (Yetidancer) [inline src/Components/RoomForm/RoomForm.js:78]: is this where we put the "press enter to add rule" functionality?

File: src/Components/RoomForm/RoomForm.js [hunk 3]
```diff
@@ -7,33 +7,40 @@ function RoomForm() {
...
   const [ potentialRule, setRule ] = useState('')
-  const roomID = uniqid()
-  // useEffect()
+  const [ roomID, setroomID ] = useState(uniqid())
+
+  const messagesEndRef = React.createRef()
+
+  const scrollToBottom = () => {
+    messagesEndRef.current.scrollIntoView({ behavior: 'smooth' })
+  }
```
Comment (Yetidancer) [inline src/Components/RoomForm/RoomForm.js:16]: what does createRef do?

File: src/Components/JoinForm/JoinForm.js [hunk 4]
```diff
@@ -8,38 +8,42 @@ function JoinForm() {
...
-            <button>
+          </section>
+          <section className='button-box'>
+            <Link to='/create'>
+              <button>
+              Create Group
+              </button>
+            </Link>
+            <Link to='/chat'>
+              <button disabled={ !(nickname && roomCode) }>
```
Comment (Yetidancer) [inline src/Components/JoinForm/JoinForm.js]: these looks like semantic changes and mostly structural

File: src/Components/JoinForm/JoinForm.js [hunk 5]
```diff
@@ -8,38 +8,42 @@ function JoinForm() {
...
-              id='name-input'
-              type='text'
-              placeholder='Name'
-              value={ nickname }
-              onChange={(e) => { setNickname(e.target.value) }}
+            id='name-input'
+            type='text'
+            placeholder='Name'
+            value={ nickname }
+            onChange={(e) => { setNickname(e.target.value) }}
```
Comment (Yetidancer) [inline src/Components/JoinForm/JoinForm.js]: just getting rid of an indentation?

File: src/Components/RoomForm/RoomForm.js (hunk 2 above)
Comment (KmBarnett) [reply src/Components/RoomForm/RoomForm.js:78]: yes handle enter checks that the enter key was the target and handles it the same as clicking the +button would do

File: src/Components/App/App.js [hunk 6]
```diff
@@ -3,17 +3,25 @@ import './App.css';
 import JoinForm from '../JoinForm/JoinForm.js';
 import RoomForm from '../RoomForm/RoomForm.js';
 import { Route } from 'react-router-dom';
+import {ReactComponent as ChatLogo} from './assets/message.svg'
 
 function App() {
   return (
-    <div className="App">
+    <main className="App">
```
Comment (Yetidancer) [inline src/Components/App/App.js]: div/vs main--why is this change necessary?

File: src/Components/RoomForm/RoomForm.js (hunk 3 above)
Comment (KmBarnett) [reply src/Components/RoomForm/RoomForm.js:16]: https://reactjs.org/docs/refs-and-the-dom.html this is a great article, but basically it creates a reference point on the page for use with:
* Managing focus, text selection, or media playback.
* Triggering imperative animations.
* Integrating with third-party DOM libraries.

File: src/Components/App/App.js (hunk 6 above)
Comment (KmBarnett) [reply src/Components/App/App.js]: Semantic for screen readers

File: src/index.js [hunk 7]
```diff
@@ -3,11 +3,14 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './Components/App/App';
 import * as serviceWorker from './serviceWorker';
+import { BrowserRouter as Router } from 'react-router-dom'
```
Comment (KmBarnett) [inline src/index.js:6]: Good Job implementing router

Comment (Yetidancer) [reply src/index.js:6]: why THANK YOU kyle

File: src/Components/RoomForm/RoomForm.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import './RoomForm.css';
+import { Link } from 'react-router-dom';
+import uniqid from 'uniqid';
+
+function RoomForm() {
+  const [ roomname, setRoomname ] = useState('')
+  const [ rules, handleRules ] = useState([])
```
Comment (KmBarnett) [inline src/Components/RoomForm/RoomForm.js:8]: Good hooks

File: src/Components/RoomForm/RoomForm.js (hunk 3 above)
Comment (Yetidancer) [reply src/Components/RoomForm/RoomForm.js:16]: WOW I'll take a look at this

File: src/Components/RoomForm/RoomForm.js [hunk 9]
```diff
@@ -39,47 +39,45 @@ function RoomForm() {
...
+          {createRules()}
+        </section>
+        <label>
+          <input id='rule-input'
+            type='text'
+            placeholder='Rule'
+            value={ potentialRule }
+            onChange={ (e) => { setRule(e.target.value) } }
+            onKeyUp={ (e) => { handleEnter(e) } }
+          />
```
Comment (KmBarnett) [inline src/Components/RoomForm/RoomForm.js:79]: Good jobs with the adding rule functionality

Comment (Yetidancer) [reply src/Components/RoomForm/RoomForm.js:79]: learning slowly but surely!
